Source/ASDisplayNode.mm

```diff
@@ -2868,25 +2872,54 @@ - (ASInterfaceState)interfaceState
 }
 
 - (void)setInterfaceState:(ASInterfaceState)newState
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  if ([[ASCATransactionQueue sharedQueue] disabled]) {
+    [self applyPendingInterfaceState:newState];
+  } else {
+    ASDN::MutexLocker l(__instanceLock__);
+    if (_pendingInterfaceState != newState) {
+      _pendingInterfaceState = newState;
+      [[ASCATransactionQueue sharedQueue] enqueue:self];
+    }
+  }
```

Comment:
Remove the lock at the top of this method. It isn't needed and `applyPendingInterfaceState` asserts that we aren't locked. (`_unlocked_applyPendingInterfaceState:`?)

Reply:
Yes, this is a mis-merge, thanks for spotting that.